src/compose.rs

```diff
@@ -160,3 +162,117 @@ where
         }
     }
 }
+
+/// Replaces the pixels in the `bottom` image with the pixels from the top image starting from the
+/// given `(x, y)` coordinates in the `bottom` image and starting from `(0, 0)` in the `top` image.
+///
+/// # Panics
+///
+/// - If `x >= bottom.width()`
+/// - If `y >= bottom.height()`
+///
+/// # Examples
+/// ```
+/// use imageproc::compose::replace;
+/// use imageproc::gray_image;
+///
+/// let bottom = gray_image!(
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0);
+///
+/// let top = gray_image!(
+///     1, 1;
+///     1, 1;
+///     1, 1);
+///
+/// let replaced = replace(&bottom, &top, 3, 1);
+///
+/// assert_eq!(replaced, gray_image!(
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 1, 1, 0;
+///     0, 0, 0, 1, 1, 0;
+///     0, 0, 0, 1, 1, 0;
+///     0, 0, 0, 0, 0, 0;
+///     0, 0, 0, 0, 0, 0));
+/// ```
+pub fn replace<P>(bottom: &Image<P>, top: &Image<P>, x: u32, y: u32) -> Image<P>
+where
+    P: Pixel,
+{
+    let mut bottom = bottom.clone();
+    replace_mut(&mut bottom, top, x, y);
+    bottom
+}
+#[doc=generate_mut_doc_comment!("replace")]
+pub fn replace_mut<P>(bottom: &mut Image<P>, top: &Image<P>, x: u32, y: u32)
```

Comment:
Isn’t this just https://docs.rs/image/latest/image/trait.GenericImage.html#method.copy_from with a different name?

Reply:
I guess the error type on the `GenericImage` function is a bit clunky.

Reply:
I didn't see `copy_from()`, I think the functions are similar but being based on `GenericImage` may also have performance issues as we've found previously.

Also technically it is an image processing operation so belongs here more than in `image` but I don't feel too strongly about that.